src/OpenTelemetry/Trace/SpanSdk.cs

```diff
@@ -333,7 +333,16 @@ public override void SetAttribute(string key, object value)
                         new EvictingQueue<KeyValuePair<string, object>>(this.tracerConfiguration.MaxNumberOfAttributes);
                 }
 
-                this.attributes.Add(new KeyValuePair<string, object>(key ?? string.Empty, sanitizedValue));
+                var attribute = this.attributes.FirstOrDefault(a => a.Key == (key ?? string.Empty));
```

Comment:
@eddynaka Looks like the same block more or less 4 times, should we have a helper method? private void AddOrReplaceAttribute or something?

Reply:
Sure! thanks for the suggestion!